terminator.go

```diff
@@ -140,6 +141,10 @@ func (t *Terminator) waitShutdown(appCtx context.Context) {
 			go func(f Hook) {
 				defer runWg.Done()
 
+				if f.preStop > 0 {
```

Comment:
`time.Sleep` will be called for every registered hook despite the initial intention to wait once before executing all hooks

Reply:
Is it a good idea to introduce global pre-stop sleep instead of per-hook one to resolve this?

Reply:
IMO yes, that might be the point of having that for each hook? What are the use-cases ?